### PR TITLE
Integration tests refactoring

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
## Description

Refactor of webdav trait to be more readable and clear.
Use of initial slash '/' as standard way for dealing with files and folders in gherkin.
## Motivation and Context

There is confusion about how the code works.
https://github.com/owncloud/core/pull/26098#issuecomment-247564045
## How Has This Been Tested?

Running tests locally.
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: build/integration/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -22,6 +24,38 @@ public function usingDavPath($davPath) {
...
+		$this->usingOldDavPath = false;
+	}
+
+	/*public function getDavFilesPath($user){
+		if ($this->usingOldDavPath === true){
+			return $this->davPath . '/';
+		} else {
+			return $this->davPath . '/files/' . $user . '/';
+		}
+	}*/
```
Comment (PVince81) [inline build/integration/features/bootstrap/WebDav.php]: remove commented out code ?

File: build/integration/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -32,8 +66,13 @@ public function getFilesPath($user){
 		return $basePath;
 	}
 
-	public function makeDavRequest($user, $method, $path, $headers, $body = null){
-		$fullUrl = substr($this->baseUrl, 0, -4) . $this->davPath . "$path";
+	public function makeDavRequest($user, $method, $path, $headers, $body = null, $type = "files"){
+		if ( $type === "files" ){
+			echo "user: " . $user . " method: " . $method . " path: " . $path . " type: " . $type;
```
Comment (PVince81) [inline build/integration/features/bootstrap/WebDav.php]: did you use `echo` just for debugging ? maybe remove it

File: build/integration/features/dav-v2.feature [hunk 3]
```diff
@@ -3,24 +3,24 @@ Feature: dav-v2
...
-		When Downloading file "/files/user0/welcome.txt" with range "bytes=51-77"
+		When Downloading file "/welcome.txt" with range "bytes=51-77"
 		Then Downloaded content should be "example file for developers"
 
 	Scenario: Downloading a file on the new endpoint should serve security headers
-		Given using dav path "remote.php/dav/files/admin/"
+		Given using new dav path
 		And As an "admin"
-		When Downloading file "welcome.txt"
+		When Downloading file "/welcome.txt"
```
Comment (PVince81) [inline build/integration/features/dav-v2.feature]: I'm not sure we really need to enforce that, unless we want this level of precision.
In PHP it's easy to force add a slash without double by doing something like `'/' . ltrim($path, '/')`

**Review by PVince81 — COMMENTED** [review]
Looks good to me, up to you if you want to act on my comments. :+1:

File: build/integration/features/dav-v2.feature (hunk 3 above)
Comment (SergioBertolinSG) [inline build/integration/features/dav-v2.feature]: I want to have it consistent and use slash before always, having it different is confusing.

File: build/integration/features/bootstrap/WebDav.php (hunk 1 above)
Comment (SergioBertolinSG) [inline build/integration/features/bootstrap/WebDav.php]: Removed.

File: build/integration/features/bootstrap/WebDav.php (hunk 2 above)
Comment (SergioBertolinSG) [inline build/integration/features/bootstrap/WebDav.php]: Removed.

**Comment by PVince81** [comment]
Looks good, merging

**Pull request merged**